api/src/main/java/org/openmrs/Diagnosis.java

```diff
@@ -0,0 +1,171 @@
+ * in an encounter to have the same rank.(As compared to previous EMRAPI implementation, 1=PRIMARY, 2=SECONDARY)
+ * ConditionVerificationStatus shows if the diagnosis is PROVISIONAL or CONFIRMED 
+ **/
+public class Diagnosis extends BaseChangeableOpenmrsData{
+	
+	private Integer diagnosisId;
+	
+	private Encounter encounter;
+	
+	private CodedOrFreeText diagnosis;
+
+	/**
+	 * Condition 
+	 * 
+	 * May refer to a new condition to be created as a result of this diagnosis,
+	 * or an existing condition that this encounter is following up.
+	 * 
+	 * 
+	 **/
+	private Condition condition;
+
+	private ConditionVerificationStatus certainty;
+	
+	private Integer rank;
+	
+	public Diagnosis() {
+	}
+
+	/**
+	 * Diagnosis constructor with encounter, CodedOrFreeText, id, ConditionVerificationStatus, and rank
+	 *
+	 * @param encounter the encounter during which the diagnosis was made
+	 * @param diagnosis the CodedOrFreeText of the diagnosis
+	 * @param certainty the ConditionVerificationStatus of the diagnosis
+	 * @param rank the rank of the diagnosis
+	 */
+	public Diagnosis(Encounter encounter, CodedOrFreeText diagnosis, ConditionVerificationStatus certainty, Integer rank) {
+		this.encounter = encounter;
+		this.diagnosis = diagnosis;
+		this.certainty = certainty;
+		this.rank = rank;
+	}
+	
+	/**
+	 * Returns the diagnosisId.
+	 * 
+	 * @return diagnosisId
+	 */
+	@Override
+	public Integer getId() {
+		return diagnosisId;
+	}
+
+	/**
+	 * Sets diagnosis diagnosisId
+	 * 
+	 * @param diagnosisId
+	 */
+	@Override
+	public void setId(Integer diagnosisId) {
+		this.diagnosisId = diagnosisId;
+	}
+
+	/**
+	 * Returns the encounter associated with this diagnosis.
+	 * 
+	 * @return encounter
+	 */
+	public Encounter getEncounter() {
+		return encounter;
+	}
+
+	/**
+	 * Sets the diagnosis encounter associated with this diagnosis.
+	 * 
+	 * @param encounter
+	 */
+	public void setEncounter(Encounter encounter) {
+		this.encounter = encounter;
+	}
+
+	/**
+	 * Returns the CodedOrFreeText.
+	 * 
+	 * @return diagnosis
+	 */
+	public CodedOrFreeText getDiagnosis() {
+		return diagnosis;
+	}
+
+	/**
+	 * Sets the CodedOrFreeText diagnosis.
+	 * 
+	 * @param diagnosis
+	 */
+	public void setDiagnosis(CodedOrFreeText diagnosis)  {
+		this.diagnosis = diagnosis;
+	}
+
+	/**
+	 * Returns the diagnosis certainty.
+	 * 
+	 * @return certainty
+	 */
+	public ConditionVerificationStatus getCertainty() {
+		return certainty;
+	}
+
+	/**
+	 * Sets the diagnosis certainty
+	 * 
+	 * @param certainty
+	 */
+	public void setCertainty(ConditionVerificationStatus certainty)  {
+		this.certainty = certainty;
+	}
+
+	/**
+	 * Returns the diagnosis rank.
+	 * 
+	 * @return rank
+	 */
+	public Integer getRank() {
+		return rank;
+	}
+
+	/**
+	 * Sets diagnosis rank
+	 * 
+	 * @param rank
+	 */
+	public void setRank(Integer rank)   {
+		this.rank = rank;
+	}
+
+	/**
+	 * Returns the diagnosis condition.
```

Comment:
Take a look at this for an example of such a javadoc comment https://github.com/openmrs/openmrs-core/commit/f5132a1dda4a6a8fcbfe06a32dcbbe4b7a0a270d